src/js_sync.hpp

```diff
@@ -119,11 +119,14 @@ void UserClass<T>::is_admin(ContextType ctx, ObjectType object, ReturnValue &ret
 template<typename T>
 void UserClass<T>::create_user(ContextType ctx, FunctionType, ObjectType this_object, size_t argc, const ValueType arguments[], ReturnValue &return_value) {
     validate_argument_count(argc, 3, 5);
+    SyncUserIdentifier userIdentifier {
+        (std::string)Value::validated_to_string(ctx, arguments[1], "identity"),
```

Comment:
copied from here https://github.com/realm/realm-js/pull/1221/files/09c0390220dd881053dd93b86e5375acf8ef5324#diff-59d314428ce563ce4238ac4e1295e609L125

the new version of get_user made the compiler pick the wrong overload. 

probably not needed for SyncUserIdentifier